src/main/java/org/gridsuite/shortcircuit/server/repositories/ShortCircuitAnalysisResultRepository.java

```diff
@@ -31,16 +29,33 @@ public class ShortCircuitAnalysisResultRepository {
 
     private ResultRepository resultRepository;
 
-    private final ObjectMapper objectMapper = RestTemplateConfig.objectMapper();
-
     public ShortCircuitAnalysisResultRepository(GlobalStatusRepository globalStatusRepository,
                                                 ResultRepository resultRepository) {
         this.globalStatusRepository = globalStatusRepository;
         this.resultRepository = resultRepository;
     }
 
-    private static ResultEntity toResultEntity(UUID resultUuid, String result) {
-        return new ResultEntity(resultUuid, result);
+    private static ShortCircuitAnalysisResultEntity toResultEntity(UUID resultUuid, ShortCircuitAnalysisResult result) {
+        List<FaultResultEntity> faultResults = result.getFaultResults().stream().map(ShortCircuitAnalysisResultRepository::toFaultResultEntity).collect(Collectors.toList());
+        return new ShortCircuitAnalysisResultEntity(resultUuid, LocalDateTime.now(), faultResults);
+    }
+
+    private static FaultResultEntity toFaultResultEntity(FaultResult faultResult) {
+        Fault fault = faultResult.getFault();
```

Comment:
Use shortcircuit.server.dto.Fault here if we keep the shortcircuit-server's dto ?

Reply:
The method is used to convert powsybl shortcircuit results to entities to be stored in database. That's why we use the powsybl version of the results here, and not the dto implementation.